plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,247 @@
+{
+    private static final Logger log = Logger.get(DeltaLakeTableMetadataScheduler.class);
+
+    private static final String TRINO_LAST_TRANSACTION_VERSION = "trino_last_transaction_version";
+    private static final String TRINO_METADATA_SCHEMA_STRING = "trino_metadata_schema_string";
+    private static final int MAX_FAILED_COUNTS = 10;
+
+    private final DeltaLakeTableOperationsProvider tableOperationsProvider;
+    private final TypeManager typeManager;
+    private final int tableParameterLengthLimit;
+    private final int storeTableMetadataThreads;
+    private final Map<SchemaTableName, UpdateInfo> updateInfos = new ConcurrentHashMap<>();
+    private final boolean enabled;
+
+    private ExecutorService executor;
+    private ScheduledExecutorService scheduler;
+    private final AtomicInteger failedCounts = new AtomicInteger();
+
+    @Inject
+    public DeltaLakeTableMetadataScheduler(
+            DeltaLakeTableOperationsProvider tableOperationsProvider,
+            NodeManager nodeManager,
+            TypeManager typeManager,
+            @TableParameterLengthLimit int tableParameterLengthLimit,
+            DeltaLakeConfig config)
+    {
+        this.tableOperationsProvider = requireNonNull(tableOperationsProvider, "tableParameterLengthLimit is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.tableParameterLengthLimit = tableParameterLengthLimit;
+        this.storeTableMetadataThreads = config.getStoreTableMetadataThreads();
+        requireNonNull(nodeManager, "nodeManager is null");
+        this.enabled = config.isStoreTableMetadataEnabled() && nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    public void putAll(Map<SchemaTableName, UpdateInfo> tableParameters)
+    {
+        updateInfos.putAll(tableParameters);
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (enabled) {
+            executor = storeTableMetadataThreads == 0 ? newDirectExecutorService() : newFixedThreadPool(storeTableMetadataThreads, threadsNamed("store-table-metadata-%s"));
+            scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("store-table-metadata"));
+
+            scheduler.scheduleWithFixedDelay(() -> {
+                try {
+                    process();
+                }
+                catch (Throwable e) {
+                    log.warn(e, "Error storing table metadata");
+                }
+                try {
+                    checkFailedTasks();
+                }
+                catch (Throwable e) {
+                    log.warn(e, "Error canceling metadata update tasks");
+                }
+            }, 200, 1000, MILLISECONDS);
+        }
+    }
+
+    @VisibleForTesting
+    public void process()
+    {
+        List<Callable<Void>> tasks = new ArrayList<>();
+        synchronized (this) {
+            Map<SchemaTableName, UpdateInfo> updateTables = updateInfos.entrySet().stream()
```

Comment:
Can you pls add a test case with the following operations to ensure that the batching of the operations will not cause that the information about the metadata stored on the metastore is false:
o1 `COMMENT ON TABLE mytable IS 'mytable comment o1'` 
o2 `INSERT ON mytable values (.....)`

What will likely be cached is the version 6 of the table

Reply:
Generally, I feel that the PR tests single statements in isolation without focusing much on multiple operations which may be run in parallel and cause a batching of the informations stored in the metastore.

`TestDeltaLakeConnectorTest` can be used however for such purposes.

Reply:
> What will likely be cached is the version 6 of the table

Where does the '6' come from?

Reply:
Left-over. :) 

I was trying to come up with a scenario where a table comment might get possibly lost. 
I don't think though that this is the case because we get the comments from the commited transaction log so no danger here.